src/main/java/entities/User.java

```diff
@@ -0,0 +1,133 @@
+package entities;
+
+public class User {
+    // ArrayList of User's favourited posts
+    private ArrayList<Post> favourites;
+
+    // ArrayList of Posts User has made
+    private ArrayList<Post> posts;
+
+    // boolean relating if User has admin privileges
+    private boolean isAdmin;
+
+    // Number of Users created
+    public static int numUsers = 0;
+
+    // id is the unique identifier of User, equal to number of posts at time of creation
+    private int id;
+
+    /*
+    * Initializes an instance of User
+    *
+    * @param admin whether the user will have admin priviledges
+    * */
+    public User(boolean admin) {
```

Comment:
You could add a parameter for the number of users in the system since we do not want this number to restart at 0 every time the program is re-run. So we would probably save the number of users in the database and, when creating a new one, get that data from the DB and feed it as a parameter into this constructor and then assign the new user the id = number of users + 1 (and then increment number of users by 1). What do you think? This is the way I implemented the posts.

Reply:
Looked into this and it seems that class variables don't persist with databases, so I'll switch mine over to your implementation